### PR TITLE
[Sprint 49] XD-3054 Validate when processing deployment message

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add deployment validation when the deployment message is consumed
by the deployment message consumer. Previously this check was only when
the deployment message is published.

### CODE REVIEW

File: spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/AbstractDeployer.java [hunk 1]
```diff
@@ -89,11 +89,13 @@ protected AbstractDeployer(ZooKeeperConnection zkConnection, PagingAndSortingRep
 	@Override
 	public D save(D definition) {
 		Assert.notNull(definition, "Definition may not be null");
+		String name = definition.getName();
+		String def = definition.getDefinition();
+		validateBeforeSave(name, def);
 		if (repository.findOne(definition.getName()) != null) {
```
Comment (pperalta) [inline spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/AbstractDeployer.java]: Doesn't `validateBeforeSave` do the same thing as:
```
if (repository.findOne(definition.getName()) != null) {
	throwDefinitionAlreadyExistsException(definition);
}
```

File: spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/AbstractInstancePersistingDeployer.java [hunk 2]
```diff
@@ -67,6 +68,7 @@ protected void beforeDelete(D definition) {
 	@Override
 	public void undeploy(String name) {
 		Assert.hasText(name, "name cannot be blank or null");
+		validateBeforeUndeploy(name);
```
Comment (pperalta) [inline spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/AbstractInstancePersistingDeployer.java]: Same here, it seems that `validateBeforeUndeploy` is doing the exact same thing as the code below.

**Comment by ilayaperumalg** [comment]
@pperalta Pushed the changes that remove the duplicate validation.

**Comment by pperalta** [comment]
LGTM; squashed, rebased, and merged as 9161736a78a59ed082227da661fc7e5c375a63ec

**Pull request closed without merging**